nullaway/src/main/java/com/uber/nullaway/handlers/AbstractFieldContractHandler.java

```diff
@@ -223,20 +246,31 @@ protected boolean validateAnnotationSyntax(
    * @param name Name of the field.
    * @return The field with the given name, or {@code null} if the field cannot be found
    */
-  public static @Nullable VariableElement getInstanceFieldOfClass(
+  public static @Nullable VariableElement getFieldOfClass(
       Symbol.ClassSymbol classSymbol, String name) {
     Preconditions.checkNotNull(classSymbol);
     for (Element member : getEnclosedElements(classSymbol)) {
-      if (member.getKind().isField() && !member.getModifiers().contains(Modifier.STATIC)) {
+      if (member.getKind().isField()) {
         if (member.getSimpleName().toString().equals(name)) {
           return (VariableElement) member;
         }
       }
     }
     Symbol.ClassSymbol superClass = (Symbol.ClassSymbol) classSymbol.getSuperclass().tsym;
     if (superClass != null) {
-      return getInstanceFieldOfClass(superClass, name);
+      return getFieldOfClass(superClass, name);
     }
     return null;
   }
+
+  public boolean isStaticThisAnnotationField(Symbol.ClassSymbol classSymbol, String fieldName) {
```

Comment:
Renames:
```suggestion
  public boolean isThisDotStaticField(Symbol.ClassSymbol classSymbol, String expression) {
```
I suggest `expression` since `fieldName` might not be just a field name but could start with `this.`.

Also can this method be private?

Reply:
@msridhar did the name change but we can't have this as private since we are using it `RequiresNonNullHandler` to build the access path only if it is not `this.` annotation. Changed it to protected though.